pgwatch2/sql/config_store/metric_definitions.sql

```diff
@@ -6929,8 +6929,10 @@ select
   coalesce(plugin, 'physical')::text as plugin,
   active,
   case when active then 0 else 1 end as non_active_int,
-  pg_wal_lsn_diff(pg_current_wal_lsn(), restart_lsn)::int8 as restart_lsn_lag_b,
-  pg_wal_lsn_diff(pg_current_wal_lsn(), confirmed_flush_lsn)::int8 as confirmed_flush_lsn_lag_b,
+  case when pg_is_in_recovery() = FALSE then pg_wal_lsn_diff(pg_current_wal_lsn(), restart_lsn)::int8
+  else pg_wal_lsn_diff(pg_last_wal_replay_lsn(), restart_lsn)::int8 end as restart_lsn_lag_b,
+  case when pg_is_in_recovery() = FALSE then pg_wal_lsn_diff(pg_current_wal_lsn(), confirmed_flush_lsn)::int8
+  else pg_wal_lsn_diff(pg_last_wal_replay_lsn(), confirmed_flush_lsn)::int8 end as confirmed_flush_lsn_lag_b,
```

Comment:
```suggestion
  case when not pg_is_in_recovery() then pg_wal_lsn_diff(pg_current_wal_lsn(), restart_lsn)::int8
  else pg_wal_lsn_diff(pg_last_wal_replay_lsn(), restart_lsn)::int8 end as restart_lsn_lag_b,
  case when not pg_is_in_recovery() then pg_wal_lsn_diff(pg_current_wal_lsn(), confirmed_flush_lsn)::int8
  else pg_wal_lsn_diff(pg_last_wal_replay_lsn(), confirmed_flush_lsn)::int8 end as confirmed_flush_lsn_lag_b,
```

Reply:
Excellent! I did test and It works even more faster.